src/mo_frontend/typing.ml

```diff
@@ -1368,9 +1385,8 @@ and infer_exp'' env exp : T.typ =
         "cannot infer type of forward field reference %s"
         id.it
     | t ->
-      Option.iter
-        (warn env exp.at "M0154" "field %s is deprecated:\n%s" id.it)
-        (T.lookup_val_deprecation id.it tfs);
+      if not env.pre then
```

Comment:
Is the purpose of `not env.pre` to avoid some false cascading warnings/errors when there exist other type errors? (Just curious, as the condition is new for this deprecation warning.)

Reply:
I noticed we were getting duplicate warnings without this, probably because the warning was generated on both the pre and checking passes.